shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/main/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilder.java

```diff
@@ -33,7 +33,9 @@
     
     @Override
     public SingleTableRule build(final SchemaRulesBuilderMaterials materials, final SingleTableRuleConfiguration config, final Collection<ShardingSphereRule> builtRules) {
-        return new SingleTableRule(materials.getDatabaseType(), materials.getDataSourceMap(), builtRules, materials.getProps());
+        SingleTableRule singleTableRule = new SingleTableRule(materials.getDatabaseType(), materials.getDataSourceMap(), builtRules, materials.getProps());
+        config.getDefaultDataSource().ifPresent(singleTableRule::setDefaultDataSource);
```

Comment:
@lanchengx Can we use the constructor to replace the setDefaultDataSource method?